src/components/CircularProgressBar/CircularProgressBar.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from 'react';
+import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
+
+import 'react-circular-progressbar/dist/styles.css';
+
+type Props = {
+  percentage: number;
+  className?: string;
+};
+
+export function CircularProgressBar({ className, percentage }: Props) {
+  const [progress, setProgress] = useState<number>(0);
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (progress < percentage) {
+        setProgress(progress + 1);
```

Comment:
I don't think a React State should handle animations, it makes the component to re-render unnecessary times which could impact performance. If there's no other way to do it (due to time constraints) we can keep it, but just food for thoughts.